frontend/src/app/core/exercise.service.ts

```diff
@@ -243,10 +251,15 @@ export class ExerciseService {
                         !client.isInWaitingRoom
                 ),
                 switchMap((client) => this.store.select(selectVisibleVehicles)),
+                withLatestFrom(this.store.select(selectCurrentTime)),
                 pairwise(),
                 takeUntil(this.stopNotifications$)
             )
-            .subscribe(([oldVehicles, newVehicles]) => {
+            .subscribe(([[oldVehicles, oldTime], [newVehicles, newTime]]) => {
```

Comment:
I was hesitant to use 0 here because in my mind debouncing by 0ms means just not debouncing (even if it's interpreted different and working fine), so I used a timer for 1ms to debounce since readding a vehicle multiple times in a span of 1ms should also not trigger a message.

Reply:
I would put it like this: 
The `0` confuses an inexperienced JS-developer, while the `1` confuses an experienced JS-developer.

If there is a `0`, I know it is just skipping multiple synchronous emits. If there is a `1`, I think this is about latency and could potentially break on different systems, etc.